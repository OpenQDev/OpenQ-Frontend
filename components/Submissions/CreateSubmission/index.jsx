import React, { useContext, useState } from 'react';
import StoreContext from '../../../store/Store/StoreContext';

import { marked } from 'marked';

const CreateSubmission = ({ bounties, githubRepository }) => {
  console.log(bounties, 'my bounties');
  const [writingState, setWritingState] = useState('Write');
  const [title, setTitle] = useState('');
  const [selectedBounties, setSelectedBounties] = useState([]);
  const [challenges, setChallenges] = useState('');
  const [challengesHTML, setChallengesHTML] = useState('');
  const [technologiesUsed, setTechnologiesUsed] = useState('');
  const [technologiesUsedHTML, setTechnologiesUsedHTML] = useState('');
  const [problems, setProblems] = useState('');
  const [problemHTML, setProblemsHTML] = useState('');
  const [overview, setOverview] = useState('');
  const [overviewHTML, setOverviewHTML] = useState('');
  const [links, setLinks] = useState([]);
  const [videoDemo, setVideoDemo] = useState('');
  const [logo, setLogo] = useState('');
  const [coverImage, setCoverImage] = useState('');
  const [teamName, setTeamName] = useState('');
  const [appState] = useContext(StoreContext);
  const handleChawngeChallenges = (e) => {
    setChallenges(e.target.value);
    setChallengesHTML(marked.parse(e.target.value));
  };
  const handleChangeTechnologiesUsed = (e) => {
    setTechnologiesUsed(e.target.value);
    setTechnologiesUsedHTML(marked.parse(e.target.value));
  };
  const handleChangeProblems = (e) => {
    setProblems(e.target.value);
    setProblemsHTML(marked.parse(e.target.value));
  };
  const handleChangeOverview = (e) => {
    setOverview(e.target.value);
    setOverviewHTML(marked.parse(e.target.value));
  };
  const handleChangeTitle = (e) => {
    setTitle(e.target.value);
  };
  const handleWrite = () => {
    setWritingState('Write');
  };

  const handlePreview = () => {
    setWritingState('Preview');
  };
  const handleCheck = (id) => {
    if (selectedBounties.includes(id)) {
      setSelectedBounties(selectedBounties.filter((item) => item !== id));
    } else {
      setSelectedBounties([...selectedBounties, id]);
    }
  };
  const handleChangeLinks = (e) => {
    const links = e.target.value.split(',');
    setLinks(links ?? []);
  };

  const handleSubmit = async () => {
    await appState.openQPrismaClient.createSubmission({
      title,
      challengesHTML,
      technologiesUsedHTML,
      problemHTML,
      overviewHTML,
      videoDemo,
      links,
      bountyAddresses: selectedBounties,
      repositoryId: githubRepository.id,
      logo,
      coverImage,
      teamName,
    });
  };

  const selectedClasses = ' border-web-gray rounded-t-sm ';
  return (
    <div className='w-full relative flex-1 pr-16 min-w-[260px]'>
      <h2 className='text-3xl pt-4'>Create Submission</h2>
      <div className='text-muted text-sm py-2 border-b border-web-gray'>Create a submission to this hackathon</div>
      {bounties.map((bounty) => (
        <div key={bounty.address}>
          <input
            type='checkbox'
            onClick={() => {
              handleCheck(bounty.address);
            }}
            value={selectedBounties.includes(bounty.address)}
          />
          {bounty.title}
        </div>
      ))}
      <div className='border text-sm border-web-gray mt-4 w-full p-2 rounded-sm flex flex-col gap-3'>
        <input onChange={handleChangeTitle} value={title} placeholder='Title' className='input-field block w-full' />

        <input
          onChange={(e) => setTeamName(e.target.value)}
          value={teamName}
          placeholder='Team Name'
          className='input-field block w-full'
        />

        <input
          onChange={handleChangeLinks}
          value={links.join('')}
          placeholder='Links'
          className='input-field block w-full'
        />
        <input
          onChange={(e) => setVideoDemo(e.target.value)}
          value={videoDemo}
          placeholder='Video Demo Link'
          className='input-field block w-full'
        />
        <input
          onChange={(e) => setLogo(e.target.value)}
          value={logo}
          placeholder='Logo'
          className='input-field block w-full'
        />
        <input
          onChange={(e) => setCoverImage(e.target.value)}
          value={coverImage}
          placeholder='Cover Image'
          className='input-field block w-full'
        />

        <div className='flex -mx-2  px-2 border-b border-web-gray'>
          <button
            onClick={handleWrite}
            className={`pt-1  border-t border-x border-transparent w-20 ${writingState === 'Write' && selectedClasses}`}
          >
            <div className={`${writingState === 'Write' ? 'pb-5 -mb-2  bg-dark-mode' : 'p-3 '} pt-2 px-3`}>Write</div>
          </button>
          <button
            onClick={handlePreview}
            className={`pt-1  border-t border-x border-transparent w-20 ${
              writingState === 'Preview' && selectedClasses
            }`}
          >
            <div className={`${writingState === 'Preview' ? 'pb-5 -mb-2  bg-dark-mode' : 'p-3'} pt-2 px-3`}>
              Preview
            </div>
          </button>
        </div>
        {writingState === 'Write' && (
          <textarea
            value={overview}
            onChange={handleChangeOverview}
            placeholder='Overview'
            className='input-field w-full h-40'
          />
        )}
        {writingState === 'Preview' && (
          <div className='overflow-y-scroll h-32'>
            <div className='w-full h-40 p-1 markdown-body' dangerouslySetInnerHTML={{ __html: overviewHTML }}></div>
          </div>
        )}
        {writingState === 'Write' && (
          <textarea
            value={problems}
            onChange={handleChangeProblems}
            placeholder='Problems solved'
            className='input-field w-full h-32'
          />
        )}
        {writingState === 'Preview' && (
          <div className='overflow-y-scroll h-32'>
            <div className='w-full h-32 p-1 markdown-body' dangerouslySetInnerHTML={{ __html: problemHTML }}></div>
          </div>
        )}
        {writingState === 'Write' && (
          <textarea
            value={challenges}
            onChange={handleChawngeChallenges}
            placeholder='Development Challenges'
            className='input-field w-full h-32'
          />
        )}
        {writingState === 'Preview' && (
          <div className='overflow-y-scroll h-32'>
            <div className='w-full  p-1 markdown-body' dangerouslySetInnerHTML={{ __html: challengesHTML }}></div>
          </div>
        )}
        {writingState === 'Write' && (
          <textarea
            value={technologiesUsed}
            onChange={handleChangeTechnologiesUsed}
            placeholder='Technologies Used'
            className='input-field w-full h-32'
          />
        )}
        {writingState === 'Preview' && (
          <div className='overflow-y-scroll h-32'>
            <div className='w-full p-1 markdown-body' dangerouslySetInnerHTML={{ __html: technologiesUsedHTML }}></div>
          </div>
        )}
        <button onClick={handleSubmit} className='btn-primary w-40 self-end'>
          Create Submission
        </button>
      </div>
    </div>
  );
};

export default CreateSubmission;
