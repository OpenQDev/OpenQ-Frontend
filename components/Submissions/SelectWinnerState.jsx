export const RESTING = 'RESTING';
export const CONFIRM = 'CONFIRM';
export const APPROVE = 'APPROVE';
export const APPROVING = 'APPROVING';
export const TRANSFERRING = 'TRANSFERRING';
export const SUCCESS = 'SUCCESS';
export const ERROR = 'ERROR';
