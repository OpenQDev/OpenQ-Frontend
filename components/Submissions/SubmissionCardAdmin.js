import React from 'react';
import WinnerSelectAmounts from './WinnerSelectAmounts';

const SubmissionCardAdmin = ({ bounty, pr, refreshBounty }) => {
  const claimedArr = bounty.claims?.map((claim) => parseInt(claim.tier));
  const payoutAndIndex = bounty.payoutSchedule.map((payout, index) => {
    const claimed = claimedArr?.includes(index);
    return {
      claimed,
      index,
      payout,
    };
  });

  const linkedPrize = bounty?.claims?.filter((claim) => claim.claimantAsset === pr.url)[0];
  return (
    <div className='border-web-gray border-t px-2'>
      <h4 className='py-4 text-center w-full font-medium text-xl'>Select Winner</h4>
      <div className='flex flex-col w-full relative h-48 overflow-y-auto'>
        <>
          {payoutAndIndex.map((payout, index) => {
            return (
              <WinnerSelectAmounts
                pr={pr}
                disabled={linkedPrize}
                bounty={bounty}
                prize={payout}
                key={index}
                refreshBounty={refreshBounty}
              />
            );
          })}
        </>
      </div>
    </div>
  );
};

export default SubmissionCardAdmin;
