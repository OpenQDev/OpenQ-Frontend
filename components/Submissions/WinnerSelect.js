import React, { useContext, useEffect, useState } from 'react';
import useWeb3 from '../../hooks/useWeb3';
import StoreContext from '../../store/Store/StoreContext';
import LoadingIcon from '../Loading/ButtonLoadingIcon';
import { RESTING, CONFIRM, TRANSFERRING, SUCCESS, ERROR } from '../FundBounty/ApproveFundState';
import ToolTip from '../Utils/ToolTipNew';
import ModalDefault from '../Utils/ModalDefault';

const WinnerSelect = ({ prize, bounty, refreshBounty, numberOfPayouts, pr }) => {
  const [showModal, setShowModal] = useState();
  const [selectionState, setSelectionState] = useState(RESTING);
  const height = (100 / numberOfPayouts) * (numberOfPayouts - prize.index);
  const tierIndex = parseInt(prize.index);
  const [appState] = useContext(StoreContext);
  const { account, library } = useWeb3();
  const [user, setUser] = useState({});
  const [closer, setCloser] = useState('');
  const [error, setError] = useState({});
  let unit;
  useEffect(async () => {
    const userId = pr.author.id;
    if (library) {
      const closer = await appState.openQClient.getAddressById(library, userId, appState.logger);

      setCloser(closer);
    }
    setUser({ id: pr.author.id, login: pr.author.login });
  }, [library, pr]);
  const claimBounty = async () => {
    try {
      setSelectionState(TRANSFERRING);
      const transaction = await appState.openQClient.claimBounty(
        library,
        bounty.bountyAddress,
        closer,
        pr.url,
        tierIndex,
        user.id,
        user.login
      );
      if (transaction) {
        setSelectionState(SUCCESS);
        if (typeof refreshBounty === 'function') {
          refreshBounty();
        }
      }
    } catch (err) {
      appState.logger.error(err, account, bounty.id);
      const { message, title } = appState.openQClient.handleError(err, {
        bounty,
      });
      setError({ message, title });
      setSelectionState(ERROR);
    }
  };
  if (bounty.payoutTokenAddress) {
    unit = ` ${appState.tokenClient.getToken(bounty.payoutTokenAddress).name}`;
  }

  const suffixed = appState.utils.handleSuffix(tierIndex + 1);
  const saturation = tierIndex % 2 ? 84 - tierIndex : 84 - tierIndex + 1;
  const lightness = !(tierIndex % 2) ? 48 + tierIndex : 48 + tierIndex - 1;
  const hue = 400 - tierIndex * 67;
  const selectWinner = () => {
    if (prize.claimed) {
      return;
    }
    setSelectionState(CONFIRM);
    setShowModal(true);
  };
  const resetState = () => {
    setSelectionState(CONFIRM);
    setShowModal();
  };
  const goBackBtn = {
    CONFIRM: (
      <button className=' btn-danger' onClick={resetState}>
        {' '}
        Go Back{' '}
      </button>
    ),
  };

  const confirmBtn = {
    CONFIRM: (
      <ToolTip
        hideToolTip={closer}
        disabled={!closer}
        toolTipText='User has not registered their github with a wallet address, please have them register a wallet address before paying out.'
      >
        <button className={closer ? 'btn-primary' : 'btn-default'} onClick={claimBounty}>
          Confirm
        </button>
      </ToolTip>
    ),
    TRANSFERRING: (
      <button onClick={() => resetState()} className='btn-default'>
        Close
      </button>
    ),
    ERROR: (
      <button onClick={() => resetState()} className='btn-default'>
        Close
      </button>
    ),
    SUCCESS: (
      <button onClick={() => resetState()} className='btn-default'>
        Close
      </button>
    ),
  };
  const modalTitle = {
    CONFIRM: `Choosing ${suffixed} Place`,
    TRANSFERRING: `Transferring Funds...`,
    SUCCESS: 'Winner Selected and Payed!',
    ERROR: error.title,
  };
  return (
    <>
      <button
        data-testid='winnerSelect'
        onClick={selectWinner}
        disabled={prize.claimed}
        className={`flex justify-center hover:scale-110 ${
          prize.claimed ? 'cursor-not-allowed' : 'cursor-pointer hover:scale-200'
        } text-black content-center items-center w-full`}
        style={{
          height: `${height}px`,
          transform: `translateY(${100 - height}px)`,
          transformOrigin: 'left center',
          backgroundColor: !prize.claimed ? `hsl(${hue}, ${saturation}%, ${lightness}%)` : '#4f4f4f',
        }}
      >
        <div> {tierIndex + 1}</div>
        <div id='mike' className='fixed inset-0'></div>
      </button>
      {showModal && (
        <ModalDefault
          footerLeft={goBackBtn[selectionState]}
          footerRight={confirmBtn[selectionState]}
          resetState={() => {}}
          title={modalTitle[selectionState]}
          setShowModal={setShowModal}
        >
          {selectionState === CONFIRM && (
            <>
              <p className='my-2'>
                You are about to select{' '}
                <a href={pr.url} className='underline'>
                  {pr.title}
                </a>{' '}
                as {suffixed} place for the{' '}
                <a className='underline' href={bounty.url}>
                  {bounty.title}
                </a>{' '}
                challenge.
              </p>
              <p className='my-2'>
                This will automaticaly send {prize.payout}
                {bounty.bountyType === '2' ? '% of funds' : unit} staked on this competition to the author of this
                submission.
              </p>
            </>
          )}
          {selectionState === TRANSFERRING && (
            <div className='flex items-center gap-2'>
              Your request is being processed... <LoadingIcon />
            </div>
          )}
          {selectionState === ERROR && <p>{error.message}</p>}
          {selectionState === SUCCESS && (
            <>
              <p className='my-2'>
                {prize.payout}
                {bounty.bountyType === '2' ? '% of funds' : unit} staked on this competition have been sent to{' '}
                {pr.author.name || pr.author.login} at {closer.slice(0, 4)}...{closer.slice(39)} .
              </p>
            </>
          )}
        </ModalDefault>
      )}
    </>
  );
};
export default WinnerSelect;
