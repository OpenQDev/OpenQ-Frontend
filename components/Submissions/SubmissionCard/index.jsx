import React, { useContext } from 'react';
import SubmissionCardAdmin from '../SubmissionCardAdmin';
import Link from 'next/link';
import Image from 'next/image';
import SubmissionWinner from '../SubmissionWinner';
import useWeb3 from '../../../hooks/useWeb3';
import StoreContext from '../../../store/Store/StoreContext';
import WinnerSelectAmounts from '../WinnerSelectAmounts';

const SubmissionCard = ({ submission, bounty, refreshBounty }) => {
  console.log(submission, 'my submissions');
  const [appState] = useContext(StoreContext);
  const { accountData } = appState;
  const { account } = useWeb3();
  const admin = bounty && bounty?.issuer?.id === account?.toLowerCase();
  const firstUser = submission.users.nodes[0];
  const tierWon = bounty?.tierWinners?.indexOf(firstUser.github);
  const submissionTime = Math.floor(new Date(submission.createdAt).getTime() / 1000);
  const tierClaimed = bounty?.claims?.some((claim) => claim.tier == tierWon);

  const classifyTime = (time) => {
    if (time < 3600) {
      return parseInt(time / 60) + ` minute${parseInt(time / 60) > 1 ? 's' : ''} ago`;
    } else if (time < 86400) {
      return parseInt(time / 3600) + ` hour${parseInt(time / 3600) > 1 ? 's' : ''} ago`;
    } else if (time < 604800) {
      return parseInt(time / 86400) + ` day${parseInt(time / 86400) > 1 ? 's' : ''} ago`;
    } else {
      return 'More than a week ago';
    }
  };

  const linkedPrize = tierWon >= 0 && tierWon + 1;

  const bodyTextString = submission.overviewHTML
    .replace(/<\/?[^>]+(>|$)/g, '')
    .replace(/<\/?[^>]+(>|$)/g, '')
    .replace(/\s+/g, ' ')
    .trim();

  const firstSentenceFromPrBodyText = (bodyTextString) => {
    // Split the cleaned text into sentences
    const sentences = bodyTextString.match(/[^\.!\?]+[\.!\?]+/g);

    // Get the first sentence (or first 10 words if there are no sentences)
    return sentences ? sentences.slice(0, 1).join(' ') : bodyTextString.split(' ').slice(0, 100).join(' ');
  };
  const sentence = firstSentenceFromPrBodyText(bodyTextString);

  return (
    <div className={`min-w-[300px] w-60  border rounded-sm border-border-gray bg-menu-bg`}>
      <div
        target='_blank'
        rel='noopener norefferer'
        className={`flex flex-col p-6 items-center text-[0.8rem] tracking-wider placeholder-input-gray outline-none rounded-sm w-full h-120 mb-1`}
      >
        <div className='flex justify-end w-full items-center -mt-2 relative pt-2'></div>
        <div className='pt-2'>
          <div className='w-16 h-16 relative'>
            <Image
              className='rounded-full'
              src={submission?.coverImage}
              placeholder={'blur'}
              blurDataURL={'/diverse/placeholder-px.png'}
              alt='n/a'
              width={64}
              height={64}
              priority={true}
            />
          </div>
        </div>
        <Link href={`/hackathons/${bounty.repoId}/${submission.id}`} className='w-full'>
          <div className='pt-5 text-center cursor-pointer underline w-full font-medium text-xl truncate'>
            {submission.title}
          </div>
        </Link>
        <div className='text-center pt-2 text-gray-400 w-full'>
          By{' '}
          <Link href={`/user/${firstUser.id}`} className='underline cursor-pointer'>
            {firstUser.username}
          </Link>
        </div>
        <div className='text-center btn-default my-2'>{classifyTime(Date.now() / 1000 - submissionTime)}</div>

        <div className='text-center pt-2 text-gray-400 w-full'>
          On {appState.utils.formatDate(parseInt(submission.createdAt) * 1000)}
          {console.log(submission.createdAt, 'submission.createdAt')}
        </div>

        <div className=' pt-2 text-gray-400 h-20 w-full break-word'>
          {sentence}
          {'...'}
        </div>
        <Link href={submission?.url ?? ''} target='_blank' rel='noreferrer'>
          <div className='pt-4 cursor-pointer' id={'github-link'}>
            <svg xmlns='http://www.w3.org/2000/svg' width='24' height='24' viewBox='0 0 24 24' fill='white'>
              <path d='M12 0c-6.626 0-12 5.373-12 12 0 5.302 3.438 9.8 8.207 11.387.599.111.793-.261.793-.577v-2.234c-3.338.726-4.033-1.416-4.033-1.416-.546-1.387-1.333-1.756-1.333-1.756-1.089-.745.083-.729.083-.729 1.205.084 1.839 1.237 1.839 1.237 1.07 1.834 2.807 1.304 3.492.997.107-.775.418-1.305.762-1.604-2.665-.305-5.467-1.334-5.467-5.931 0-1.311.469-2.381 1.236-3.221-.124-.303-.535-1.524.117-3.176 0 0 1.008-.322 3.301 1.23.957-.266 1.983-.399 3.003-.404 1.02.005 2.047.138 3.006.404 2.291-1.552 3.297-1.23 3.297-1.23.653 1.653.242 2.874.118 3.176.77.84 1.235 1.911 1.235 3.221 0 4.609-2.807 5.624-5.479 5.921.43.372.823 1.102.823 2.222v3.293c0 .319.192.694.801.576 4.765-1.589 8.199-6.086 8.199-11.386 0-6.627-5.373-12-12-12z' />
            </svg>
          </div>
        </Link>
      </div>
      {!linkedPrize ? (
        admin && <SubmissionCardAdmin refreshBounty={refreshBounty} submission={submission} bounty={bounty} />
      ) : (
        <>
          <SubmissionWinner linkedPrize={linkedPrize} bounty={bounty} />
          {bounty?.creatingUser?.id === accountData.id && (
            <WinnerSelectAmounts
              submission={submission}
              disabled={false}
              bounty={bounty}
              refreshBounty={refreshBounty}
              isRemove={true}
              tierClaimed={tierClaimed}
              prize={{ index: tierWon, payout: bounty.payoutSchedule[tierWon] }}
            />
          )}
        </>
      )}
    </div>
  );
};
export default SubmissionCard;
