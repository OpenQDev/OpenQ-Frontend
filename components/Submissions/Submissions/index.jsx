import React from 'react';
import SubmissionCard from '../SubmissionCard';

const Submissions = ({ bounty, refreshBounty }) => {
  const submissions = bounty?.submissions?.nodes;
  console.log(bounty, 'bounty');
  return (
    <>
      <div className='grid gap-8 w-full pt-8 justify-between justify-items-center grid-cols-[repeat(auto-fit,_minmax(300px,_1fr))]'>
        {submissions?.map((submission, index) => {
          return <SubmissionCard refreshBounty={refreshBounty} key={index} submission={submission} bounty={bounty} />;
        })}
      </div>
    </>
  );
};
export default Submissions;
