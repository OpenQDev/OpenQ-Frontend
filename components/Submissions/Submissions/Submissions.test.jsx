/**
 * @vi-environment jsdom
 */
import React from 'react';

import Submissions from '.';
import ShallowRenderer from 'react-test-renderer/shallow';
import Constants from '../../../test-utils/constant';

describe('Submissions', () => {
  const bounty = Constants.bounty;

  it('should render match DOM Snapshot', () => {
    const shallow = new ShallowRenderer();
    shallow.render(<Submissions bounty={bounty} pr={bounty.prs[0]} />);
    const tree = shallow.getRenderOutput();
    expect(tree).toMatchSnapshot();
  });
});
