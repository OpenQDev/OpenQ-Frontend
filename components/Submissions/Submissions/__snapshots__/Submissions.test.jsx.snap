// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Submissions > should render match DOM Snapshot 1`] = `
<React.Fragment>
  <div
    className="w-full bg-info border-info-strong border-2 p-3 rounded-sm"
  >
    Don't see your submission yet? Click 'Refresh' above to get the latest pull requests from Github.
  </div>
  <div
    className="grid gap-8 w-full pt-8 justify-between justify-items-center grid-cols-[repeat(auto-fit,_minmax(300px,_1fr))]"
  >
    <SubmissionCard
      bounty={
        {
          "address": "0x066efd87027a4cc4a21c57467f224ef98241d774",
          "assignees": [],
          "avatarUrl": "https://avatars.githubusercontent.com/u/77402538?v=4",
          "body": "body of test2",
          "bodyHTML": "<p dir=\\"auto\\">body of test2</p>",
          "bountyAddress": "0x066efd87027a4cc4a21c57467f224ef98241d774",
          "bountyId": "I_kwDOGWnnz85I9Agl",
          "bountyMintTime": "1662545344",
          "bountyTokenBalances": [
            {
              "__typename": "BountyFundedTokenBalance",
              "tokenAddress": "0x0000000000000000000000000000000000000000",
              "volume": "2000000000000000000",
            },
            {
              "__typename": "BountyFundedTokenBalance",
              "tokenAddress": "0x9fE46736679d2D9a65F0992F2272dE9f3c7fa6e0",
              "volume": "2000000000000000000",
            },
            {
              "__typename": "BountyFundedTokenBalance",
              "tokenAddress": "0x5FbDB2315678afecb367f032d93F642f64180aa3",
              "volume": "2000000000000000000",
            },
          ],
          "bountyType": "0",
          "claims": [],
          "closed": true,
          "closedAt": "2022-03-28T17:57:44Z",
          "closedEvents": [
            {
              "__typename": "ClosedEvent",
              "actor": {
                "__typename": "User",
                "avatarUrl": "https://avatars.githubusercontent.com/u/93455288?u=fd1fb04b6ff2bf397f8353eafffc3bfb4bd66e84&v=4",
                "login": "FlacoJones",
                "name": null,
                "url": "https://github.com/FlacoJones",
              },
              "id": "CE_lADOGWnnz85GjwA1zwAAAAF4vHFc",
            },
          ],
          "closerData": null,
          "createdAt": "1661767948472",
          "deposits": [
            {
              "__typename": "Deposit",
              "expiration": "1296000",
              "id": "0xe5551a3fa87d93a0c6c084d572b9e282114befc43dc68f08be6d53d13830e356",
              "receiveTime": "1662545374",
              "refundTime": null,
              "refunded": false,
              "sender": {
                "__typename": "User",
                "id": "0xf39fd6e51aad88f6f4ce6ab8827279cfffb92266",
              },
              "tokenAddress": "0x0000000000000000000000000000000000000000",
              "volume": "1000000000000000000",
            },
            {
              "__typename": "Deposit",
              "expiration": "30",
              "id": "0xb4f31aab8a1c4bfe26236729e8cd8e4abf81d63283e006b4ec677a7ce6b2871a",
              "receiveTime": "1662545373",
              "refundTime": "1662559726",
              "refunded": true,
              "sender": {
                "__typename": "User",
                "id": "0xf39fd6e51aad88f6f4ce6ab8827279cfffb92266",
              },
              "tokenAddress": "0x9fE46736679d2D9a65F0992F2272dE9f3c7fa6e0",
              "volume": "2000000000000000000",
            },
            {
              "__typename": "Deposit",
              "expiration": "1296000",
              "id": "0x7db2691e573b9c19e90f391cd3eda9ce9246666a1502f2bf87b9d47d13679bc0",
              "receiveTime": "1662545372",
              "refundTime": null,
              "refunded": false,
              "sender": {
                "__typename": "User",
                "id": "0xf39fd6e51aad88f6f4ce6ab8827279cfffb92266",
              },
              "tokenAddress": "0x5fbdb2315678afecb367f032d93f642f64180aa3",
              "volume": "1000000000000000000",
            },
          ],
          "fundingGoalTokenAddress": "0x0000000000000000000000000000000000000000",
          "id": undefined,
          "issuer": {
            "__typename": "User",
            "id": "0xf39fd6e51aad88f6f4ce6ab8827279cfffb92266",
          },
          "labels": [
            {
              "__typename": "Label",
              "color": "0075ca",
              "name": "documentation",
            },
            {
              "__typename": "Label",
              "color": "cfd3d7",
              "name": "duplicate",
            },
          ],
          "number": 8,
          "owner": "OpenQDev",
          "prs": [
            {
              "__typename": "CrossReferencedEvent",
              "assignees": [],
              "id": "CRE_kwDOGWnnz85GHUw1",
              "referencedAt": "2022-03-28T17:47:26Z",
              "source": {
                "__typename": "PullRequest",
                "author": {
                  "__typename": "User",
                  "avatarUrl": "https://avatars.githubusercontent.com/u/93455288?u=fd1fb04b6ff2bf397f8353eafffc3bfb4bd66e84&v=4",
                  "email": "",
                  "id": "U_kgDOBZIDuA",
                  "login": "FlacoJones",
                  "url": "https://github.com/FlacoJones",
                  "user": {
                    "__typename": "User",
                    "login": "FlacoJones",
                    "url": "https://github.com/FlacoJones",
                  },
                },
                "body": "This Closes #136 and also unrelated thing of Fixes #137",
                "bodyHTML": "<p dir=\\"auto\\">This <span class=\\"issue-keyword tooltipped tooltipped-se\\" aria-label=\\"This pull request closes issue #136.\\">Closes</span> <a class=\\"issue-link js-issue-link\\" data-error-text=\\"Failed to load title\\" data-id=\\"1183776821\\" data-permission-text=\\"Title is private\\" data-url=\\"https://github.com/OpenQDev/OpenQ-TestRepo/issues/136\\" data-hovercard-type=\\"issue\\" data-hovercard-url=\\"/OpenQDev/OpenQ-TestRepo/issues/136/hovercard\\" href=\\"https://github.com/OpenQDev/OpenQ-TestRepo/issues/136\\">#136</a> and also unrelated thing of <span class=\\"issue-keyword tooltipped tooltipped-se\\" aria-label=\\"This pull request closes issue #137.\\">Fixes</span> <a class=\\"issue-link js-issue-link\\" data-error-text=\\"Failed to load title\\" data-id=\\"1183777365\\" data-permission-text=\\"Title is private\\" data-url=\\"https://github.com/OpenQDev/OpenQ-TestRepo/issues/137\\" data-hovercard-type=\\"issue\\" data-hovercard-url=\\"/OpenQDev/OpenQ-TestRepo/issues/137/hovercard\\" href=\\"https://github.com/OpenQDev/OpenQ-TestRepo/issues/137\\">#137</a></p>",
                "bodyText": "This Closes #136 and also unrelated thing of Fixes #137",
                "createdAt": "2022-03-28T17:57:44Z",
                "id": "PR_kwDOGWnnz841LGsK",
                "mergeCommit": {
                  "__typename": "Commit",
                  "author": {
                    "__typename": "GitActor",
                    "avatarUrl": "https://avatars.githubusercontent.com/u/93455288?v=4",
                    "name": "FlacoJones",
                    "user": {
                      "__typename": "User",
                      "login": "FlacoJones",
                      "url": "https://github.com/FlacoJones",
                    },
                  },
                },
                "merged": true,
                "mergedAt": "2022-03-28T17:57:44Z",
                "title": "Update README.md",
                "url": "https://github.com/OpenQDev/OpenQ-TestRepo/pull/138",
              },
            },
          ],
          "refunds": [
            {
              "__typename": "Refund",
              "depositId": "0xb4f31aab8a1c4bfe26236729e8cd8e4abf81d63283e006b4ec677a7ce6b2871a",
              "refundTime": "1662559726",
              "tokenAddress": "0x9fE46736679d2D9a65F0992F2272dE9f3c7fa6e0",
              "volume": "2000000000000000000",
            },
          ],
          "repoName": "OpenQ-Frontend",
          "title": "Properly Referenced and Merged by FlacoJones",
          "titleHTML": "Properly Referenced and Merged by FlacoJones",
          "tvl": "9.51848",
          "twitterUsername": "openqlabs",
          "url": "https://github.com/OpenQDev/OpenQ-TestRepo/issues/8",
        }
      }
      pr={
        {
          "__typename": "PullRequest",
          "author": {
            "__typename": "User",
            "avatarUrl": "https://avatars.githubusercontent.com/u/93455288?u=fd1fb04b6ff2bf397f8353eafffc3bfb4bd66e84&v=4",
            "email": "",
            "id": "U_kgDOBZIDuA",
            "login": "FlacoJones",
            "url": "https://github.com/FlacoJones",
            "user": {
              "__typename": "User",
              "login": "FlacoJones",
              "url": "https://github.com/FlacoJones",
            },
          },
          "body": "This Closes #136 and also unrelated thing of Fixes #137",
          "bodyHTML": "<p dir=\\"auto\\">This <span class=\\"issue-keyword tooltipped tooltipped-se\\" aria-label=\\"This pull request closes issue #136.\\">Closes</span> <a class=\\"issue-link js-issue-link\\" data-error-text=\\"Failed to load title\\" data-id=\\"1183776821\\" data-permission-text=\\"Title is private\\" data-url=\\"https://github.com/OpenQDev/OpenQ-TestRepo/issues/136\\" data-hovercard-type=\\"issue\\" data-hovercard-url=\\"/OpenQDev/OpenQ-TestRepo/issues/136/hovercard\\" href=\\"https://github.com/OpenQDev/OpenQ-TestRepo/issues/136\\">#136</a> and also unrelated thing of <span class=\\"issue-keyword tooltipped tooltipped-se\\" aria-label=\\"This pull request closes issue #137.\\">Fixes</span> <a class=\\"issue-link js-issue-link\\" data-error-text=\\"Failed to load title\\" data-id=\\"1183777365\\" data-permission-text=\\"Title is private\\" data-url=\\"https://github.com/OpenQDev/OpenQ-TestRepo/issues/137\\" data-hovercard-type=\\"issue\\" data-hovercard-url=\\"/OpenQDev/OpenQ-TestRepo/issues/137/hovercard\\" href=\\"https://github.com/OpenQDev/OpenQ-TestRepo/issues/137\\">#137</a></p>",
          "bodyText": "This Closes #136 and also unrelated thing of Fixes #137",
          "createdAt": "2022-03-28T17:57:44Z",
          "id": "PR_kwDOGWnnz841LGsK",
          "mergeCommit": {
            "__typename": "Commit",
            "author": {
              "__typename": "GitActor",
              "avatarUrl": "https://avatars.githubusercontent.com/u/93455288?v=4",
              "name": "FlacoJones",
              "user": {
                "__typename": "User",
                "login": "FlacoJones",
                "url": "https://github.com/FlacoJones",
              },
            },
          },
          "merged": true,
          "mergedAt": "2022-03-28T17:57:44Z",
          "title": "Update README.md",
          "url": "https://github.com/OpenQDev/OpenQ-TestRepo/pull/138",
        }
      }
    />
  </div>
</React.Fragment>
`;
